webapp/components/form/AutocompleteDialog/AutocompleteDialog.js

```diff
@@ -0,0 +1,181 @@
+            focusItem(0)
+          } else {
+            close()
+          }
+
+          break
+        case KeyboardMap.Esc:
+          close()
+          inputField.focus()
+          break
+        default:
+        // Do nothing
+      }
+    },
+    [items, inputField, focusItem, close]
+  )
+
+  const calculateNewIndexWithOffset = useCallback(
+    (offset) => {
+      const itemsSize = (items || []).length
+      let index = focusedItemIndex + offset
+      if (index < 0) {
+        index = 0
+      } else if (index >= itemsSize) {
+        index = itemsSize - 1
+      }
+      return index
+    },
+    [items, focusedItemIndex]
+  )
+
+  const onListItemKeyDown = useCallback(
+    (event) => {
+      event.stopPropagation()
+      event.preventDefault()
+      const itemsSize = (items || []).length
+
+      if (itemsSize > 0) {
+        switch (event.keyCode) {
+          case KeyboardMap.Enter:
+          case KeyboardMap.Space:
+            selectItem(items[focusedItemIndex])
+            break
+          case KeyboardMap.Esc:
+            close()
+            inputField.focus()
+            break
+          case KeyboardMap.PageUp:
+          case KeyboardMap.PageDown:
+          case KeyboardMap.Up:
+          case KeyboardMap.Down:
+            focusItem(calculateNewIndexWithOffset(offsetByKey[event.keyCode]))
+            break
+          default:
+          // Do nothing
+        }
+      }
+    },
+    [items, inputField, close, focusItem, focusedItemIndex, selectItem, calculateNewIndexWithOffset]
+  )
+
+  useEffect(() => {
+    if (inputField) {
+      inputField.addEventListener('keydown', onInputFieldKeyDown)
+    }
+    window.addEventListener('click', onOutsideClick)
+
+    return () => {
+      if (inputField) {
+        inputField.removeEventListener('keydown', onInputFieldKeyDown)
+      }
+
+      window.removeEventListener('click', onOutsideClick)
+    }
+  }, [inputField, onInputFieldKeyDown, onOutsideClick])
+
+  return (
+    <div ref={list} className={`autocomplete-list ${className}`} style={{ ...calculatePosition() }}>
+      {(items || []).map((item) => (
+        <ItemRenderer
+          key={itemKeyFunction(item)}
+          tabIndex="0"
+          item={item}
+          itemLabel={itemLabel}
+          onKeyDown={onListItemKeyDown}
+          onMouseDown={selectItem}
+        />
+      ))}
+    </div>
+  )
+}
+
+AutocompleteDialog.propTypes = {
+  inputField: PropTypes.any,
+  sourceElement: PropTypes.any,
+  items: PropTypes.array,
+  itemRenderer: PropTypes.any,
+  itemLabel: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
+  itemKeyFunction: PropTypes.func,
+  className: PropTypes.string,
+  onItemSelect: PropTypes.func,
+  onClose: PropTypes.func,
+}
+
+AutocompleteDialog.defaultProps = {
+  items: [],
+  itemRenderer: null,
+  itemLabel: null, // required prop / string or function
+  itemKeyFunction: null, // required prop / string or function - Rename it to itemKey
```

Comment:
as you can see from this comment, this should be renamed to itemKey and be oneOfType string or function